### PR TITLE
[FLINK-17832][docs][es] Add documentation for the new Elasticsearch connector

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

Add documentation for the new Elasticsearch connector.

## Brief change log

- Added `elasticsearch.md` for Table/SQL connectors
- Update links in `connectors/formats/index.md` and `connectors/index.md`.

## Verifying this change

N/A

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/table/connectors/elasticsearch.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+under the License.
+-->
+
+<span class="label label-primary">Sink: Batch</span>
+<span class="label label-primary">Sink: Streaming Append & Upsert Mode</span>
+
+* This will be replaced by the TOC
+{:toc}
+
+The Elasticsearch connector allows for writing into an index of the Elasticsearch search engine. This document describes how to setup the Elasticsearch Connector to run SQL queries against Elasticsearch.
```
Comment (godfreyhe) [inline docs/dev/table/connectors/elasticsearch.md:32]: simply `Elasticsearch search engine.`  to `Elasticsearch engine.` ?

File: docs/dev/table/connectors/elasticsearch.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+The connector can operate in upsert mode for exchanging UPDATE/DELETE messages with the external system using the primary key defined on the DDL.
+
+If no primary key is defined on the DDL, the connector can only operate in append mode for exchanging INSERT only messages with external system.
+
+Dependencies
+------------
+
+In order to setup the Elasticsearch connector, the following table provide dependency information for both projects using a build automation tool (such as Maven or SBT) and SQL Client with SQL JAR bundles.
```
Comment (godfreyhe) [inline docs/dev/table/connectors/elasticsearch.md:41]: nit: provide -> provides

File: docs/dev/table/connectors/elasticsearch.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    </tbody>
+</table>
+
+Features
+----------------
+
+### Key Handling
+
+Elasticsearch sink can work in either upsert mode or append mode, it depends on whether primary key is defined.
+If primary key is defined, Elasticsearch sink works in upsert mode which can consume queries contain UPDATE/DELETE messages.
```
Comment (godfreyhe) [inline docs/dev/table/connectors/elasticsearch.md:240]: contain -> containing

File: docs/dev/table/connectors/elasticsearch.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+If primary key is not defined, Elasticsearch sink works in append mode which can only consume queries contain INSERT only messages.
```
Comment (godfreyhe) [inline docs/dev/table/connectors/elasticsearch.md:241]: ditto

File: docs/dev/table/connectors/elasticsearch.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+      <td><h5>document-type</h5></td>
+      <td>required in 6.x</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>Elasticsearch document type. Not necessary anymore in <code>elasticsearch-7</code>.</td>
+    </tr>
+    <tr>
+      <td><h5>document-id.key-delimiter</h5></td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">_</td>
```
[the remark this answers is not in the document]
Comment (wuchong) [reply docs/dev/table/connectors/elasticsearch.md:125]: Yes. I added the comment in the description column.

File: docs/dev/table/connectors/elasticsearch.zh.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+You can also use `'{field_name|date_format_string}'` to convert a field value of `TIMESTAMP/DATE/TIME` type into the format specified by the `date_format_string`.
+The `date_format_string` is compatible with Java's [DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/index.html).
+For example, if the option value is `'myusers-{log_ts|yyyy-MM-dd}'`, then a record with `log_ts` field value `2020-03-27 12:25:55` will be written into "myusers-2020-03-27" index.
+
+
+Data Type Mapping
+----------------
+
+Elasticsearch stores document in a JSON string. So the data type mapping is between Flink data type and JSON data type.
+Flink uses built-in `'json'` format for Elasticsearch connector. Please refer to <a href="{% link dev/table/connectors/formats/index.md %}">JSON Format</a> page for more type mapping details.
```
Comment (danny0405) [inline docs/dev/table/connectors/elasticsearch.zh.md:266]: We better link to the json page directly.

Comment (wuchong) [reply docs/dev/table/connectors/elasticsearch.zh.md:266]: You are right. The json page is not available when I was writing this page.

Comment (wuchong) [reply docs/dev/table/connectors/elasticsearch.zh.md:266]: I will correct all the links in a PR.

**Pull request closed without merging**